components/Navigation/index.js

```diff
@@ -22,40 +22,93 @@ var SECONDARY = [
   ['events', 'Events'],
 ];
 
-var renderNavigation = (list, props) => (
+var renderNavigation = (list, scope) => (
   <ul className="Navigation-list">
     {list.map(n => (
       <li className="Navigation-item" key={n[0]}>
-        <Link className="Navigation-link" to={n[0]} onClick={props.onClick}>{n[1]}</Link>
+        <Link className="Navigation-link" to={n[0]} onClick={scope.toggleOverlay}>{n[1]}</Link>
       </li>
     ))}
   </ul>
 );
 
 class Navigation extends Component {
+
+  constructor(props: mixed, context: mixed): void {
+    super(props, context);
+    this.state = {showNav: true, atTop: true, overlayVisible: false};
+    this.previousScrollPos = 0;
+
+    this.toggleOverlay = this.toggleOverlay.bind(this);
+  }
+
+  componentDidMount(): void {
+    if (typeof window === 'undefined') return;
+    if (window.scrollY) {
+      this.setState({atTop: false});
+    }
+    window.addEventListener('scroll', _ => {
+      var showNav = true;
+      var atTop = false;
+
+      if (this.previousScrollPos - window.scrollY < 0) {
+        showNav = false;
+      } else {
+        showNav = true;
+      }
+
+      if (window.scrollY < 10) {
+        showNav = true;
+        atTop = true;
+      }
+
+      this.previousScrollPos = window.scrollY;
+      this.setState({showNav: showNav, atTop: atTop});
+
+    }, false);
+  }
+
+  componentWillUnmount(): void {
+    if (typeof window === 'undefined') return;
+    window.removeEventListener('scroll');
+  }
+
+  toggleOverlay(): void {
+    if (!this.state.overlayVisible) {
+      document.getElementsByTagName('body')[0].className = 'noscroll';
+    } else {
+      document.getElementsByTagName('body')[0].className = '';
+    }
+    this.setState({overlayVisible: !this.state.overlayVisible});
+  }
+
   render(): ?ReactElement {
-    var {visible} = this.props;
-    var className = `Navigation-mobile ${visible ? 'is-visible' : 'is-not-visible'}`;
+    var mobileClassName = `Navigation-mobile ${this.state.overlayVisible ? 'is-visible' : 'is-not-visible'}`;
+    var mainClass = `Navigation-main ${this.state.showNav ? 'is-visible' : 'is-not-visible'}`;
+    mainClass += this.state.atTop ? ' is-top' : '';
+    var hamburgerClass = `Navigation-hamburger ${this.state.showNav ? 'is-visible' : 'is-not-visible'}`;
```

Comment:
you could probably do the `var somethingSomething = this.state.showNav ? 'is-visible' : 'is-not-visible'` and not do so many if else’s